meshroom/ui/qml/GraphEditor/GraphEditor.qml

```diff
@@ -118,19 +118,31 @@ Item {
     }
 
     Keys.onPressed: {
-        if (event.key === Qt.Key_F)
-            fit()
-        if (event.key === Qt.Key_Delete)
-            if (event.modifiers == Qt.AltModifier)
-                uigraph.removeNodesFrom(uigraph.selectedNodes)
-            else
-                uigraph.removeNodes(uigraph.selectedNodes)
-        if (event.key === Qt.Key_D)
-            duplicateNode(event.modifiers == Qt.AltModifier)
-        if (event.key === Qt.Key_C && event.modifiers == Qt.ControlModifier)
-            copyNodes()
-        if (event.key === Qt.Key_V && event.modifiers == Qt.ControlModifier)
-            pasteNodes()
+        if (event.key === Qt.Key_F) {
+            fit();
+        } 
+        else if (event.key === Qt.Key_Delete) {
+            if (event.modifiers == Qt.AltModifier) {
+                uigraph.removeNodesFrom(uigraph.selectedNodes);
+            }
+            else {
+                uigraph.removeNodes(uigraph.selectedNodes);
+            }
+        }
+        else if (event.key === Qt.Key_D) {
+            duplicateNode(event.modifiers == Qt.AltModifier);
+        }
+        else if (event.key === Qt.Key_C && event.modifiers == Qt.ControlModifier) {
+            copyNodes();
+        }
+        else if (event.key === Qt.Key_V && event.modifiers == Qt.ControlModifier) {
+            pasteNodes();
+        }
+        else if (event.key == Qt.Key_Tab) {
+            event.accepted = true;
+            newNodeMenu.spawnPosition = mouseArea.mapToItem(draggable, mouseArea.mouseX, mouseArea.mouseY);
```

Comment:
Check if the mouse position is in the GraphEditor and if it's not inside, the node creation should be done in the center.

Reply:
See https://github.com/alicevision/Meshroom/pull/1788

Reply:
I don't think the menu should appear if the GraphEditor doesn't have the focus: the "tab" key is ambiguous, it is used by default to select one element after another outside the GraphEditor (and it is the "classic" behaviour for that key). 

If the node creation menu appears in the center of the GraphEditor even when it doesn't have the focus, then this behaviour will be overridden.